test/functions/coerce.js

```diff
@@ -110,13 +110,58 @@ test('coerce tests', (t) => {
     ['1.2.3/6', '6.0.0', { rtl: true }],
     ['1.2.3.4', '2.3.4', { rtl: true }],
     ['1.2.3.4xyz', '2.3.4', { rtl: true }],
+
+    ['1-rc.5', '1.0.0-rc.5', { includePrerelease: true }, true],
+    ['1.2-rc.5', '1.2.0-rc.5', { includePrerelease: true }, true],
+    ['1.2.3-rc.5', '1.2.3-rc.5', { includePrerelease: true }, true],
+    ['1.2.3-rc.5/a', '1.2.3-rc.5', { includePrerelease: true }, true],
+    ['1.2.3.4-rc.5', '1.2.3', { includePrerelease: true }, true],
+    ['1.2.3.4+rev.6', '1.2.3', { includePrerelease: true }, true],
+
+    ['1+rev.6', '1.0.0+rev.6', { includePrerelease: true }, true],
+    ['1.2+rev.6', '1.2.0+rev.6', { includePrerelease: true }, true],
+    ['1.2.3+rev.6', '1.2.3+rev.6', { includePrerelease: true }, true],
+    ['1.2.3+rev.6/a', '1.2.3+rev.6', { includePrerelease: true }, true],
+    ['1.2.3.4-rc.5', '1.2.3', { includePrerelease: true }, true],
+    ['1.2.3.4+rev.6', '1.2.3', { includePrerelease: true }, true],
+
+    ['1-rc.5+rev.6', '1.0.0-rc.5+rev.6', { includePrerelease: true }, true],
+    ['1.2-rc.5+rev.6', '1.2.0-rc.5+rev.6', { includePrerelease: true }, true],
+    ['1.2.3-rc.5+rev.6', '1.2.3-rc.5+rev.6', { includePrerelease: true }, true],
+    ['1.2.3-rc.5+rev.6/a', '1.2.3-rc.5+rev.6', { includePrerelease: true }, true],
+
+    ['1.2-rc.5+rev.6', '1.2.0-rc.5+rev.6', { rtl: true, includePrerelease: true }, true],
+    ['1.2.3-rc.5+rev.6', '1.2.3-rc.5+rev.6', { rtl: true, includePrerelease: true }, true],
+    ['1.2.3.4-rc.5+rev.6', '2.3.4-rc.5+rev.6', { rtl: true, includePrerelease: true }, true],
+    ['1.2.3.4-rc.5', '2.3.4-rc.5', { rtl: true, includePrerelease: true }, true],
+    ['1.2.3.4+rev.6', '2.3.4+rev.6', { rtl: true, includePrerelease: true }, true],
+    ['1.2.3.4-rc.5+rev.6/7', '7.0.0', { rtl: true, includePrerelease: true }, true],
+    ['1.2.3.4-rc/7.5+rev.6', '7.5.0+rev.6', { rtl: true, includePrerelease: true }, true],
+    ['1.2.3.4/7-rc.5+rev.6', '7.0.0-rc.5+rev.6', { rtl: true, includePrerelease: true }, true],
   ]
-  coerceToValid.forEach(([input, expected, options]) => {
-    const msg = `coerce(${input}) should become ${expected}`
-    t.same((coerce(input, options) || {}).version, expected, msg)
+  coerceToValid.forEach(([input, expected, options, shouldParse]) => {
+    const coerceExpression = `coerce(${input}, ${JSON.stringify(options)})`
+    const coercedVersion = coerce(input, options) || {}
+    let expectedText = expected
+
+    if (shouldParse) {
```

Comment:
This is surprising to me.  What is different about the new fixtures that is not present in the old ones that means we only do this to the new ones? What is this asserting specifically?  If this is important I would assume that all of the tests would want this assertion.

Reply:
Would any of the existing fixtures fail this test? If not we can remove that flag altogether and run them all through both assertions.  Having that extra step if it's not needed is extra cognitive load for the folks who come after us trying to understand the tests.